daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -458,53 +413,94 @@ class CLIConf(arguments: Array[String]) extends scallop.ScallopConf(arguments)
-    val config = opt[File](short = 'c', argName = "file", descr = "path to file containing configuration items.")
+    val config = opt[File](short = 'c', argName = "file", descr = "Path to XML file containing configuration items")
+    val vars = props[String]('D', keyName = "variable", valueName = "value", descr = "Variables to be used when parsing. An optional namespace may be provided.")
+    val path = opt[String](argName = "path", descr = "Path from root element to node from which to start parsing")
+    val rootNS = opt[RefQName]("root", argName = "node", descr = "The root element to use. An optional namespace may be provided. This must be one of the top-level elements of the DFDL schema defined with --schema. Requires --schema. If not supplied, uses schema's first top-level element.")
+    val schema = opt[URI]("schema", argName = "file", required = true, descr = "The annotated DFDL schema to use to create the parser")(fileResourceURIConverter)
+    val tunables = props[String]('T', keyName = "tunable", valueName = "value", descr = "Tunable configuration options to change Daffodil's behavior")
+
+    val outfile = trailArg[String](required = false, descr = "Output file to save parser to. If not specified, or a value of -, saves to stdout.")
 
     requireOne(schema) // --schema must be provided
     validateFileIsFile(config) // --config must be a file that exists
   }
 
   // Test Subcommand Options
   object test extends scallop.Subcommand("test") {
-    banner("""|Usage: daffodil test <tdmlfile> [testname...]
+    banner("""|Usage: daffodil test [-l] [-r] [-i] <tdmlfile> [testname...]
               |
               |List or execute tests in a TDML file
               |
               |Test Options:""".stripMargin)
 
-    descr("list or execute TDML tests")
+    descr("List or execute TDML tests")
     helpWidth(76)
 
-    val list = opt[Boolean](descr = "show names and descriptions instead of running test cases.")
-    val regex = opt[Boolean](descr = "treat <names> as regular expressions.")
-    val tdmlfile = trailArg[String](required = true, descr = "test data markup language (TDML) file.")
-    val names = trailArg[List[String]](required = false, descr = "name of test case(s) in tdml file. If not given, all tests in tdmlfile are run.")
-    val info = tally(descr = "increment test result information output level, one level for each -i")
+    val info = tally(descr = "Increment test result information output level, one level for each -i")
+    val list = opt[Boolean](descr = "Show names and descriptions instead of running test cases")
+    val regex = opt[Boolean](descr = "Treat <testname...> as regular expressions")
+    val tdmlfile = trailArg[String](required = true, descr = "Test Data Markup Language (TDML) file")
+    val names = trailArg[List[String]](required = false, descr = "Name(s) of test cases in tdmlfile. If not given, all tests in tdmlfile are run.")
```

Comment:
The banner testname but the help is going to say "names". We should make that consistent while we are changing things.